spec/features/user_deletes_book_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+feature 'User deletes a book', js: true do
+  scenario 'successfully' do
+    book = create(:book)
+    visit root_path
+
+    expect {
+      accept_confirm t('books.delete.confirmation_text') do
+        click_on t('books.delete.link')
+      end
+    }.to change(Book, :count).by(-1)
```

Comment:
This style of test makes it a bit harder to follow what's going on and is a bit vague. It also might be good to extract a helper here. Something like this:

```ruby
delete_book

expect(page).to have_content 'Books index header' #use i18n here
expect(page).not_to have_content(book.title)
```

You may want to use `have_css` instead of `have_content` if you're doing that in other tests. That way you have a consistent style.

You can also extract a helper if you're using things in multiple tests:

```
expect(page).to have_books_index_title

def have_books_index_title
  have_content 'The i18n for the books index page
end
```

You can see the thought process behind extracting helpers here: https://www.wetestrails.com/blog/extract-matcher and https://www.varvet.com/blog/simple-tricks-to-clean-up-your-capybara-tests/ though it is probably overkill to make a custom matcher. You can just extract a private method like in my example most of the time.

You can see some great examples in the Formkeep tests. Here is an example of extracting a helper for expectations: https://github.com/thoughtbot/formkeep/blob/master/spec/support/features/plans.rb#L6-L8